src/System.Net.Sockets/tests/FunctionalTests/AgnosticListenerTest.cs

```diff
@@ -112,6 +112,33 @@ public void StaticCreate_Success()
             listener.Stop();
         }
 
+        [OuterLoop] // TODO: Issue #11345
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void AllowNatTraversal_Windows()
+        {
+            var l = new TcpListener(IPAddress.Any, 0);
+            l.AllowNatTraversal(true);
+            Assert.Equal((int)IPProtectionLevel.Unrestricted, (int)l.Server.GetSocketOption(SocketOptionLevel.IP, SocketOptionName.IPProtectionLevel));
+
+            l = new TcpListener(IPAddress.Any, 0);
+            l.AllowNatTraversal(false);
+            Assert.Equal((int)IPProtectionLevel.EdgeRestricted, (int)l.Server.GetSocketOption(SocketOptionLevel.IP, SocketOptionName.IPProtectionLevel));
+        }
+
+        [OuterLoop] // TODO: Issue #11345
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public void AllowNatTraversal_AnyUnix()
+        {
+            var l = new TcpListener(IPAddress.Any, 0);
+            Assert.Throws<PlatformNotSupportedException>(() => l.AllowNatTraversal(true));
+
+            l = new TcpListener(IPAddress.Any, 0);
+            Assert.Throws<PlatformNotSupportedException>(() => l.AllowNatTraversal(false));
+        }
```

Comment:
Nit: if you make this a Theory with two InlineDatas, one for true and one for false, you can avoid the duplication, e.g.
```C#
[OuterLoop]
[PlatformSpecific(TestPlatforms.AnyUnix)]
[Theory]
[InlineData(true)]
[InlineData(false)]
public void AllowNatTraversal_AnyUnix(bool allowed)
{
    var listener = new TcpListener(IPAddress.Any, 0);
    Assert.Throws<PlatformNotSupportedException>(() => l.AllowNatTraversal(allowed));
}
```